pkgs/development/tools/kubectx/default.nix

```
@@ -13,20 +11,15 @@ stdenv.mkDerivation rec {
     sha256 = "1b22jk8zl944w5zn3s7ybkkbmzp9519x32pfqwd1malfly7dzf55";
   };
 
-  buildInputs = [ makeWrapper ];
-
-  dontBuild = true;
-  doCheck = false;
+  goPackagePath = "github.com/ahmetb/kubectx";
+  vendorSha256 = "168hfdc2rfwpz2ls607bz5vsm1aw4brhwm8hmbiq1n1l2dn2dj0y";
 
-  installPhase = ''
+  postInstall = ''
     mkdir -p $out/bin
     mkdir -p $out/share/zsh/site-functions
     mkdir -p $out/share/bash-completion/completions
     mkdir -p $out/share/fish/vendor_completions.d
```

you can remove all the shell completion code, and replace it with:
```suggestion
  postInstall = ''
    installShellCompletion completion/*
  '';
```
still get:
```
$ tree ./result/share/
./result/share/
├── bash-completion
│   └── completions
│       ├── kubectx.bash
│       └── kubens.bash
├── fish
│   └── vendor_completions.d
│       ├── kubectx.fish
│       └── kubens.fish
└── zsh
    └── site-functions
        ├── _kubectx
        └── _kubens
```
```